### PR TITLE
[FEAT] Travelling Salesman npc

### PR DESCRIPTION
**Pull request by Rofrtd**
# Description

This PR implements the Travelling Salesman NPC which will offer special offers in exchange of resources.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Unit tests

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/farming/trader/Trader.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+          onClick={handleOpenModal}
+          className="w-full"
+        />
+      </div>
+
+      <Modal centered show={modalState !== "closed"} onHide={handleCloseModal}>
+        <Panel>
+          <img
+            src={close}
+            className="h-6 top-4 right-4 absolute cursor-pointer"
```
Comment (adamhannigan) [inline src/features/farming/trader/Trader.tsx:154]: Consider adding a z-index `z-10`, I can't seem to click this on the trade modal

Screen Shot 2022-06-21 at 10 30 35 am

**Review by adamhannigan — CHANGES_REQUESTED** [review]
Awesome work. All looks good to me!

There is just a small issue that is preventing me from closing the modal.

**Review by adamhannigan — DISMISSED** [review]
LGTM 🚀

**Review by c-py — CHANGES_REQUESTED** [review]
Looking good! Just one small change suggested.

File: src/features/announcements/index.ts [hunk 2]
```diff
@@ -23,4 +24,16 @@ export const ANNOUNCEMENTS: Announcement[] = [
     link: "https://docs.sunflower-land.com/crafting-guide/farming-and-gathering#animals",
     image: chicken,
   },
+  {
+    date: new Date(
+      "Tue Jun 17 2022 08:10:00 GMT-0300 (Brasilia Standard Time)"
+    ),
```
[the remark this answers is not in the document]
Comment (adamhannigan) [reply src/features/announcements/index.ts:30]: And what is this Brasilia Standard Time mate ;) Get with the times! 😆

**Pull request merged**